Makefile

```diff
@@ -17,18 +25,54 @@ FFMPEG_OPTS ?= --prefix=$(FFMPEG_BUILD) \
                --enable-network \
                --enable-parser=h264 \
                --enable-parser=hevc
-
 CGO_LDFLAGS := -L$(FFMPEG_BUILD)/lib
-ifeq ($(UNAME_S),Linux)
+export PKG_CONFIG_PATH=$(FFMPEG_BUILD)/lib/pkgconfig
+
+# If we are building for android, we need to set the correct flags
+# and toolchain paths for FFMPEG and go binary cross-compilation.
+# Make sure to install android SDK and NDK before building.
+# If you are using a different version of NDK, please set the NDK_VERSION variable.
+ifeq ($(TARGET_OS),android)
+# x86 android targets have not been tested, so we do not support them for now.
+ifeq ($(TARGET_ARCH),arm64)
+    # Android build doesn't support most of our cgo libraries, so we use the no_cgo flag.
+    GO_TAGS ?= -tags no_cgo
+    # We need the go build command to think it's in cgo mode to support android NDK cross-compilation.
+    export CGO_ENABLED = 1
+    NDK_VERSION ?= 26.1.10909125
+	ifeq ($(SOURCE_OS),darwin)
+        NDK_ROOT ?= $(HOME)/Library/Android/Sdk/ndk/$(NDK_VERSION)
+    else ifeq ($(SOURCE_OS),linux)
+        NDK_ROOT ?= $(HOME)/android-ndk-r26
```

Comment:
Isn't this hard coding the version through another means which doesn't use the `NDK_VERSION` variable?

Reply:
Yes good call. Adjusted so that on Macos it will fill in path based on NDK_VERSION by using ls. On, linux the path fill is much more straightforward.